src/neuroconv/nwbconverter.py

```diff
@@ -87,11 +88,19 @@ def _validate_source_data(self, source_data: dict[str, dict], verbose: bool = Fa
     def __init__(self, source_data: dict[str, dict], verbose: bool = False):
         """Validate source_data against source_schema and initialize all data interfaces."""
         self.verbose = verbose
-        self.data_interface_objects = {
-            name: data_interface(**source_data[name])
-            for name, data_interface in self.data_interface_classes.items()
-            if name in source_data
-        }
+        self.data_interface_objects = dict()
+        for interface_name, data_interface in self.data_interface_classes.items():
+            # If the name of an interface is not passed in the source_data, skip it
+            # TODO: shouldn't this throw an error? keeping the current behavior now
```

Comment:
I think this is intended, or at least I make use of this functionality fairly often. For some conversions, there are sessions that only use a subset of the interfaces. Rather than having to define multiple NWBConverters for each subset, I can just define one NWBConverter class and modify the source_data dictionary to only include the interfaces that I want.

Reply:
I see. That makes sense to me.

Reply:
I removed the TODO and added an explanation on how this is used.